service/application/src/main/java/com/solace/maas/ep/common/messages/CommandMessage.java

```diff
@@ -33,9 +38,19 @@ public CommandMessage(String serviceId,
         this.serviceId = serviceId;
         this.commandCorrelationId = commandCorrelationId;
         this.context = context;
+        this.status = status;
         this.commandBundles = commandBundles;
     }
 
+    public void determineStatus() {
+        boolean hasAtLeastOneError = false;
+        if (CollectionUtils.isNotEmpty(commandBundles)) {
+            hasAtLeastOneError = commandBundles.stream().anyMatch(bundle ->
+                    bundle.getCommands().stream().anyMatch(Command::hasSignificantErrorResult));
+        }
+        this.setStatus(hasAtLeastOneError ? JobStatus.error : JobStatus.success);
+    }
```

Comment:
Noting what I/we were thinking here like a mini ADR:
It would be ideal if this logic was on a BO and both the BO and mop message had a `status` field. Since we start with this command message and send the same thing back as a response, there's not a clear separation. Since the places we set command errors are in the application module _and_ tf plugin module that uses the BO (CommandRequest), it's less error prone & a bit cleaner to calculate the status just before we send the response.

This pojo is doing the calculation because it requires no outside data going in or out for better encapsulation.
We don't set the status first, then calculate before sending to avoid a case where we report the default status set in the beginning. If we don't set the status first and there's a bug (we don't calculate the status), we'll get null and can easily notice the issue.

Reply:
Good suggestion, I like the idea of adding a BO layer for the reasons you mention. I'll add it in.